OADP-4561: Update parent ToC for Troubleshooting modules

apurvabhide17 opened the pull request:
Version(s):
OCP 4.14 - 4.19

Issue:
[OADP-4561](https://issues.redhat.com/browse/OADP-4561)

Link to docs preview: 
- [Troubleshooting OADP](https://89097--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/troubleshooting),
- [Velero CLI tool](https://89097--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/velero-cli-tool), [Pods crash or restart due to lack of memory or CPU](https://89097--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/pods-crash-or-restart-due-to-lack-of-memory-or-cpu), 
- [Issues with Velero and admission webhooks](https://89097--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/issues-with-velero-and-admission-webhooks), 
- [OADP installation issues](https://89097--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/oadp-installation-issues),
- [OADP timeouts](https://89097--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/oadp-timeouts),
- [Backup and Restore CR issues](https://89097--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/backup-and-restore-cr-issues), 
- [Restic issues](https://89097--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/restic-issues), 
- [Using the must-gather tool](https://89097--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/using-the-must-gather-tool), 
- [OADP Monitoring](https://89097--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/oadp-monitoring) 

QE review:
- [x] QE has approved this change.

Additional information:

- This task focuses on changing the taxonomy for the [Troubleshooting](https://docs.openshift.com/container-platform/4.16/backup_and_restore/application_backup_and_restore/troubleshooting.html) user story. 
- Here, I've removed the includes of individual modules from the troubleshooting assembly, and added them to the topic.yaml so that the related modules are visible in the ToC. 
- I haven't updated any content from this user story, and there are no technical changes implemented in this PR. I've converted a few concept modules into assembly (will be shown as a new file due to this conversion), and included its related modules.

apurvabhide17 committed:
OADP-4561: Update parent ToC for Troubleshooting modules

Removed unnecessary syntax from newly added assemblies

fixed xref issue

Added xrefs of all modules to the troubleshooting assembly

Added missing xref anchor ID

Peer review

Merge review suggestions

apurvabhide17 commented:
/label OADP

anarnold97 commented on backup_and_restore/application_backup_and_restore/issues-with-velero-and-admission-webhooks.adoc:
```diff
@@ -0,0 +1,33 @@
+[role="_additional-resources"]
+.Additional resources
+
+* xref:../../architecture/admission-plug-ins.adoc[Admission plugins]
```
@apurvabhide17 - please add an anchor to the xref

anarnold97 commented on backup_and_restore/application_backup_and_restore/backup-and-restore-cr-issues.adoc:
```diff
@@ -0,0 +1,97 @@
+[id="backup-cr-remains-partiallyfailed_{context}"]
+== Backup CR status remains in PartiallyFailed
+
+The status of a `Backup` CR without Restic in use remains in the `PartiallyFailed` phase and does not complete. A snapshot of the affiliated PVC is not created.
```
```suggestion
The status of a `Backup` CR without Restic in use remains in the `PartiallyFailed` phase and is not completed. A snapshot of the affiliated PVC is not created.
```

stillalearner reviewed (approved):
lgtm

apurvabhide17 commented:
/label peer-review-needed

- This task focuses on changing the taxonomy for the [Troubleshooting](https://docs.openshift.com/container-platform/4.16/backup_and_restore/application_backup_and_restore/troubleshooting.html) user story.
- Here, I've removed the include statements of individual modules from the troubleshooting assembly, and added them to the topic.yaml so that the related modules are visible in the right-ens side ToC for docs.redhat.com.
- I haven't updated any content from this user story, and there are no technical changes implemented in this PR. I've converted a few concept modules into assembly (will be shown as a new file due to this conversion), and included its related modules.

jneczypor commented:
/label peer-review-in-progress

jneczypor commented on backup_and_restore/application_backup_and_restore/oadp-monitoring.adoc:
```diff
@@ -0,0 +1,26 @@
+
+toc::[]
+
+The {product-title} provides a monitoring stack that allows users and administrators to effectively monitor and manage their clusters, as well as monitor and analyze the workload performance of user applications and services running on the clusters, including receiving alerts if an event occurs.
```
```suggestion
The {product-title} provides a monitoring stack that allows users and administrators to effectively monitor and manage their clusters as well as monitor and analyze the workload performance of user applications and services running on the clusters, including receiving alerts if an event occurs.
```

jneczypor commented on backup_and_restore/application_backup_and_restore/pods-crash-or-restart-due-to-lack-of-memory-or-cpu.adoc:
```diff
@@ -0,0 +1,36 @@
+  memory: 128Mi
+----
+====
+
```
I believe the handbook says that multiple points or ideas in a note should be conveyed as bullets. Maybe make your two notes two bullets. You might also want to consider pulling the code out and putting it in the docs. This is a really long note.

apurvabhide17 replied:
For simplicity, removed this information from the admonition and added it before the include statement.

jneczypor commented on backup_and_restore/application_backup_and_restore/restic-issues.adoc:
```diff
@@ -0,0 +1,90 @@
+[id="restic-permission-error-nfs-root-squash-enabled_{context}"]
+== Restic permission error for NFS data volumes with root_squash enabled
+
+The `Restic` pod log displays the error message: `controller=pod-volume-backup error="fork/exec/usr/bin/restic: permission denied"`.
```
```suggestion
The `Restic` pod log displays the following error message: `controller=pod-volume-backup error="fork/exec/usr/bin/restic: permission denied"`.
```

jneczypor commented on backup_and_restore/application_backup_and_restore/restic-issues.adoc:
```diff
@@ -0,0 +1,90 @@
+
+.Solution
+
+You can resolve this issue by creating a supplemental group for `Restic` and adding the group ID to the `DataProtectionApplication` manifest:
```
```suggestion
You can resolve this issue by creating a supplemental group for `Restic` and adding the group ID to the `DataProtectionApplication` manifest.
```

jneczypor commented on backup_and_restore/application_backup_and_restore/troubleshooting.adoc:
```diff
@@ -12,149 +12,21 @@ include::_attributes/attributes-openshift-dedicated.adoc[]
-include::modules/migration-debugging-velero-resources.adoc[leveloffset=+1]
+* Collect logs and CR information by using the xref:../../backup_and_restore/application_backup_and_restore/using-the-must-gather-tool.adoc#using-the-must-gather-tool[`must-gather` tool].
 
+* Monitor and analyze the workload performance with the help of xref:../../backup_and_restore/application_backup_and_restore/oadp-monitoring.adoc#oadp-monitoring[OADP Monitoring].
```
```suggestion
* Monitor and analyze the workload performance with the help of xref:../../backup_and_restore/application_backup_and_restore/oadp-monitoring.adoc#oadp-monitoring[OADP monitoring].
```

jneczypor commented on backup_and_restore/application_backup_and_restore/using-the-must-gather-tool.adoc:
```diff
@@ -0,0 +1,122 @@
+This operation can take a long time. This command saves the data as the `must-gather/metrics/prom_data.tar.gz` file.
+endif::[]
+ifdef::oadp-troubleshooting[]
+* Full `must-gather` data collection, including Prometheus metrics:
```
Should this be a complete sentence?
```suggestion
* For full `must-gather` data collection, including Prometheus metrics, run the following command:
```

jneczypor commented on backup_and_restore/application_backup_and_restore/using-the-must-gather-tool.adoc:
```diff
@@ -0,0 +1,122 @@
++
+The data is saved as `must-gather/must-gather.tar.gz`. You can upload this file to a support case on the link:https://access.redhat.com/[Red Hat Customer Portal].
+
+* Essential `must-gather` data collection, without Prometheus metrics, for a specific time duration:
```
```suggestion
* For essential `must-gather` data collection, without Prometheus metrics, for a specific time duration, run the following command:
```

jneczypor commented on backup_and_restore/application_backup_and_restore/using-the-must-gather-tool.adoc:
```diff
@@ -0,0 +1,122 @@
+----
+<1> Specify the time in hours. Allowed values are `1h`, `6h`, `24h`, `72h`, or `all`, for example, `gather_1h_essential` or `gather_all_essential`.
+
+* `must-gather` data collection with timeout:
```
```suggestion
* For `must-gather` data collection with timeout, run the following command:
```

jneczypor commented on backup_and_restore/application_backup_and_restore/using-the-must-gather-tool.adoc:
```diff
@@ -0,0 +1,122 @@
+----
+<1> Specify a timeout value in seconds.
+
+* Prometheus metrics data dump:
```
```suggestion
* For a Prometheus metrics data dump, run the following command:
```

jneczypor commented on backup_and_restore/application_backup_and_restore/velero-cli-tool.adoc:
```diff
@@ -0,0 +1,22 @@
+
+toc::[]
+
+You can obtain the Velero CLI tool by:
```
```suggestion
You can obtain the Velero CLI tool by using one of the following options:
```

jneczypor reviewed (commented):
This is really great work, especially for such a large PR! Just some minor comments.
/remove-label peer-review-in-progress
/remove-label peer-review-needed
/label peer-review-done

apurvabhide17 commented:
> This is really great work, especially for such a large PR! Just some minor comments. /remove-label peer-review-in-progress /remove-label peer-review-needed /label peer-review-done

Thanks @jneczypor for reviewing. Most of these modules need updates to have consistency and whenever I review the modularization work for this user story, I'll work on it :)

apurvabhide17 commented:
/label merge-review-needed

xenolinux commented on backup_and_restore/application_backup_and_restore/pods-crash-or-restart-due-to-lack-of-memory-or-cpu.adoc:
```diff
@@ -0,0 +1,45 @@
+[IMPORTANT]
+====
+The values for the resource request fields must follow the same format as Kubernetes resource requirements.
+Also, if you do not specify `configuration.velero.podConfig.resourceAllocations` or `configuration.restic.podConfig.resourceAllocations`, the default `resources` specification for a Velero pod or a Restic pod is as follows:
```
```suggestion
If you do not specify `configuration.velero.podConfig.resourceAllocations` or `configuration.restic.podConfig.resourceAllocations`, see the following default `resources` specification configuration for a Velero or Restic pod:
```

xenolinux commented on backup_and_restore/application_backup_and_restore/restic-issues.adoc:
```diff
@@ -1,25 +1,28 @@
 .Solution
 
-You can resolve this issue by creating a supplemental group for `Restic` and adding the group ID to the `DataProtectionApplication` manifest:
+You can resolve this issue by creating a supplemental group for `Restic` and adding the group ID to the `DataProtectionApplication` manifes
```
```suggestion
You can resolve this issue by creating a supplemental group for `Restic` and adding the group ID to the `DataProtectionApplication` manifest:
```

- Typo

xenolinux commented on backup_and_restore/application_backup_and_restore/using-the-must-gather-tool.adoc:
```diff
@@ -0,0 +1,122 @@
+
+.Prerequisites
+
+* You must be logged in to the {product-title} cluster as a user with the `cluster-admin` role.
+* You must have the OpenShift CLI (`oc`) installed.
```
```suggestion
* You have logged in to the {product-title} cluster as a user with the `cluster-admin` role.
* You have installed the OpenShift CLI (`oc`).
```

xenolinux commented on backup_and_restore/application_backup_and_restore/velero-cli-tool.adoc:
```diff
@@ -0,0 +1,22 @@
+
+toc::[]
+
+You can obtain the Velero CLI tool by using the following options:
+
+* Downloading the Velero CLI tool
+* Accessing the Velero binary in the Velero deployment in the cluster
```
```suggestion
You can obtain the `velero` CLI tool by using the following options:

* Downloading the `velero` CLI tool
* Accessing the `velero` binary in the Velero deployment in the cluster
```

I think velero should se the monospace when being used as CLI and binary names.

xenolinux commented on backup_and_restore/application_backup_and_restore/using-the-must-gather-tool.adoc:
```diff
@@ -0,0 +1,122 @@
+$ oc adm must-gather --image={must-gather-v1-4}
+----
++
+The data is saved as `must-gather/must-gather.tar.gz`. You can upload this file to a support case on the link:https://access.redhat.com/[Red Hat Customer Portal].
```
```suggestion
The data is saved as `must-gather/must-gather.tar.gz`. You can upload this file to a support case on the link:https://access.redhat.com/[Red{nbsp}Hat Customer Portal].
```

xenolinux commented on backup_and_restore/application_backup_and_restore/using-the-must-gather-tool.adoc:
```diff
@@ -0,0 +1,122 @@
+$ oc adm must-gather --image=registry.redhat.io/rhmtc/openshift-migration-must-gather-rhel8:v1.8 -- /usr/bin/gather_metrics_dump
+----
++
+This operation can take a long time. This command saves the data as the `must-gather/metrics/prom_data.tar.gz` file.
```
```suggestion
This operation can take a long time. This command saves the data in the `must-gather/metrics/prom_data.tar.gz` file.
```

xenolinux commented on backup_and_restore/application_backup_and_restore/using-the-must-gather-tool.adoc:
```diff
@@ -0,0 +1,122 @@
+$ oc adm must-gather --image=registry.redhat.io/rhmtc/openshift-migration-must-gather-rhel8:v1.7
+----
++
+This command saves the data as the `must-gather/must-gather.tar.gz` file. You can upload this file to a support case on the link:https://access.redhat.com/[Red Hat Customer Portal].
```
```suggestion
This command saves the data as the `must-gather/must-gather.tar.gz` file. You can upload this file to a support case on the link:https://access.redhat.com/[Red{nbsp}Hat Customer Portal].
```

xenolinux commented on backup_and_restore/application_backup_and_restore/troubleshooting.adoc:
```diff
@@ -12,149 +12,21 @@ include::_attributes/attributes-openshift-dedicated.adoc[]
-include::modules/velero-obtaining-by-downloading.adoc[leveloffset=+1]
-include::modules/velero-oadp-version-relationship.adoc[leveloffset=+2]
-include::modules/velero-obtaining-by-accessing-binary.adoc[leveloffset=+1]
+* Use the available xref:../../backup_and_restore/application_backup_and_restore/oadp-timeouts.adoc#oadp-timeouts[OADP timeouts] to reduce the likelihood of errors, retries, or failures.
```
```suggestion
* Use the available xref:../../backup_and_restore/application_backup_and_restore/oadp-timeouts.adoc#oadp-timeouts[OADP timeouts] to reduce errors, retries, or failures.
```

or replace "likelihood" with "occurrence"

xenolinux reviewed (commented):
Left a few comments and nits before the merge. Overall looks good.

Please ensure the commits are squashed to 1.

apurvabhide17 commented:
/test validate-portal

apurvabhide17 commented:
> Left a few comments and nits before the merge. Overall looks good.
> 
> Please ensure the commits are squashed to 1.

Thanks, @xenolinux.

apurvabhide17 commented:
/label merge-review-needed

jab-rh commented:
@apurvabhide17, confirm this does not go into 4.19? If so, it shouldn't go into `main` branch either. Thanks!

apurvabhide17 commented:
> @apurvabhide17, confirm this does not go into 4.19? If so, it shouldn't go into `main` branch either. Thanks!

@jab-rh, added 4.19 in the description

stevsmit commented on backup_and_restore/application_backup_and_restore/using-the-must-gather-tool.adoc:
```diff
@@ -0,0 +1,122 @@
+.Procedure
+
+. Navigate to the directory where you want to store the `must-gather` data.
+. Run the `oc adm must-gather` command for one of the following data collection options:
```
In the preview, the content stops here: https://89097--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/using-the-must-gather-tool.html

None of the content below renders.

apurvabhide17 replied:
done, I've fixed it by removing the conditionals, which were not required for OADP.

stevsmit reviewed (commented):
Hey! Some smaller and larger things to look at before merging. The two larger things: 

* The page "Installation Issues" that you've shared as a preview in the description of the PR does not render. I'm not sure if this is intentional or not. 
* In the module backup_and_restore/application_backup_and_restore/using-the-must-gather-tool.adoc, the content stops after line 42, "Run the `oc adm muster-gather. . ." line. I see that the content is split by ifdefs, but I'm not sure where that content is rendering. Since it appears to me that your reorganizing this content, I'm not sure if the ifdefs used in that assembly are necessary. 

Other nits are bit smaller. You have some friend-reference xref content that needs fixed. 

Overall it looks good. Just ping me tomorrow if you've taken a look at the module.

apurvabhide17 commented on _topic_maps/_topic_map.yml:
```diff
@@ -3647,12 +3647,32 @@ Topics:
     File: oadp-advanced-topics
+  - Name: Troubleshooting OADP
+    File: troubleshooting
+  - Name: Velero CLI tool
```
@stevsmit, since these modules were previously nested under Troubleshooting, those modules were not visible in the right-side ToC. 
For the same reason, we're trying to restructure OADP docs so that the required modules are visible in the ToC, and reader can find them easily from the ToC instead of scrolling.

apurvabhide17 commented on backup_and_restore/application_backup_and_restore/oadp-monitoring.adoc:
```diff
@@ -0,0 +1,31 @@
+include::modules/oadp-monitoring-setup.adoc[leveloffset=+1]
+include::modules/oadp-creating-service-monitor.adoc[leveloffset=+1]
+include::modules/oadp-creating-alerting-rule.adoc[leveloffset=+1]
+[role="_additional-resources"]
```
done, I've fixed it in this PR itself. Thanks :)

apurvabhide17 commented on backup_and_restore/application_backup_and_restore/oadp-operator-issues.adoc:
```diff
@@ -23,23 +26,21 @@ Retrieve a list of backup storage locations (BSLs) and check the manifest of eac
 ----
 
-. Using the list of BSLs, run the following command to display the manifest of each BSL, and examine each manifest for an error.
+. Use the list of BSLs from the previous step and run the following command to examine the manifest of each BSL for an error:
```
done, I've fixed it in this PR. Thanks :)

apurvabhide17 commented on backup_and_restore/application_backup_and_restore/oadp-timeouts.adoc:
```diff
@@ -0,0 +1,36 @@
+
+* xref:../../backup_and_restore/application_backup_and_restore/oadp-timeouts.adoc#csisnapshot-timeout_oadp-timeouts[CSI snapshot timeout]
+
+* xref:../../backup_and_restore/application_backup_and_restore/oadp-timeouts.adoc#item-operation-timeout-restore_oadp-timeouts[Item operation timeout - backup]
```
good catch, I've fixed it

apurvabhide17 commented on backup_and_restore/application_backup_and_restore/oadp-timeouts.adoc:
```diff
@@ -0,0 +1,36 @@
+
+* xref:../../backup_and_restore/application_backup_and_restore/oadp-timeouts.adoc#item-operation-timeout-restore_oadp-timeouts[Item operation timeout - backup]
+
+* xref:../../backup_and_restore/application_backup_and_restore/oadp-timeouts.adoc#item-operation-timeout-backup_oadp-timeouts[Item operation timeout - restore]
```
good catch, I've fixed it

apurvabhide17 commented on backup_and_restore/application_backup_and_restore/pods-crash-or-restart-due-to-lack-of-memory-or-cpu.adoc:
```diff
@@ -0,0 +1,31 @@
+
+[role="_additional-resources"]
+.Additional resources
+* xref:../../backup_and_restore/application_backup_and_restore/installing/about-installing-oadp.adoc#oadp-velero-cpu-memory-requirements_about-installing-oadp[CPU and memory requirements]
```
good catch, I've fixed it

apurvabhide17 commented:
> Hey! Some smaller and larger things to look at before merging. The two larger things:
> 
> * The page "Installation Issues" that you've shared as a preview in the description of the PR does not render. I'm not sure if this is intentional or not.
> * In the module backup_and_restore/application_backup_and_restore/using-the-must-gather-tool.adoc, the content stops after line 42, "Run the `oc adm muster-gather. . ." line. I see that the content is split by ifdefs, but I'm not sure where that content is rendering. Since it appears to me that your reorganizing this content, I'm not sure if the ifdefs used in that assembly are necessary.
> 
> Other nits are bit smaller. You have some friend-reference xref content that needs fixed.
> 
> Overall it looks good. Just ping me tomorrow if you've taken a look at the module.
Hi @stevsmit, adding a few comments related to your above points -

- For modules nested under Troubleshooting - since all the troubleshooting-related modules were previously nested under Troubleshooting, those modules were not visible in the right-side ToC. For the same reason, you would see all these modules are now added to the topic.yaml instead of including them in the Troubleshooting assembly. 
- I've updated the link for [OADP installation issues](https://89097--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/oadp-installation-issues) in the description.
- For the must-gather module - I've removed all the conditionals related to MTC and MTA. There is a similar file named "migration-using-must-gather.adoc", which is included in MTC and MTV where all these conditionals are included.
Keeping this file simple for this PR and I'll discuss it with the team to understand if it needs any update w.r.t to the other files related to MTC and MTV. 

I've implemented other suggestions that you pointed. Could you merge this PR now?
Thank you :)

stevsmit commented:
/cherry-pick enterprise-4.16

stevsmit commented:
/cherry-pick enterprise-4.18

stevsmit commented:
/cherry-pick enterprise-4.19

stevsmit commented:
/cherry-pick enterprise-4.19

stevsmit commented:
/cherry-pick enterprise-4.18
